### ISSUE TITLE
Error when trying to use project move utility

### ISSUE THREAD

**Issue by angelicochris** [taori/Amusoft.VisualStudio.Extensions#7]
![image](https://user-images.githubusercontent.com/55987158/108231410-bcbec880-710f-11eb-805d-a4a385969698.png)

Looking up the log it mentions I found this error

```xml
<entry>
    <record>620</record>
    <time>2021/02/17 15:52:03.662</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>CreateInstance failed for package [ToolingPackage]Source: &apos;mscorlib&apos; Description: Could not load file or assembly &apos;Microsoft.VisualStudio.Shell.15.0, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a&apos; or one of its dependencies. The system cannot find the file specified.&#x000D;&#x000A;System.IO.FileNotFoundException: Could not load file or assembly &apos;Microsoft.VisualStudio.Shell.15.0, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a&apos; or one of its dependencies. The system cannot find the file specified.&#x000D;&#x000A;File name: &apos;Microsoft.VisualStudio.Shell.15.0, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a&apos;&#x000D;&#x000A;   at System.Reflection.RuntimeAssembly.GetType(RuntimeAssembly assembly, String name, Boolean throwOnError, Boolean ignoreCase, ObjectHandleOnStack type)&#x000D;&#x000A;   at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)&#x000D;&#x000A;   at System.Activator.CreateInstanceFromInternal(String assemblyFile, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityInfo)&#x000D;&#x000A;   at System.AppDomain.CreateInstanceFrom(String assemblyFile, String typeName)&#x000D;&#x000A;&#x000D;&#x000A;WRN: Assembly binding logging is turned OFF.&#x000D;&#x000A;To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.&#x000D;&#x000A;Note: There is some performance penalty associated with assembly bind failure logging.&#x000D;&#x000A;To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].&#x000D;&#x000A;</description>
    <guid>{5594EE3B-1FF9-4B15-B5DB-ED7864223937}</guid>
    <hr>80004005 - E_FAIL</hr>
    <errorinfo></errorinfo>
</entry>`

**Comment by taori** [comment]
Thanks for reporting. Might take a little to fix it because i have to figure out what is going wrong here. Most likely related to VS version not being correct. Could you tell me which version of VS you use?

**Comment by angelicochris** [comment]
> 
> 
> Thanks for reporting. Might take a little to fix it because i have to figure out what is going wrong here. Most likely related to VS version not being correct. Could you tell me which version of VS you use?

Visual Studios 2017, more specifically 15.9.33. Appreciate the quick response 👍

**Comment by taori** [comment]
That may be why i guess. I am using VS 2019. I suppose it works for me because i have 2017 installed still. Is using VS2019 an option for you?

**Comment by angelicochris** [comment]
> 
> 
> That may be why i guess. I am using VS 2019. I suppose it works for me because i have 2017 installed still. Is using VS2019 an option for you?

Not yet unfortunately my work is looking to upgrade, but there is no guarantee. Trying to make a template for our MVC projects and we have a specific structure we use with multiple projects for each thing, which of course as we know causes issues when trying to make multi project templates. I am in the process of trying to figure out the wizard method, but not having any luck so far and was going to just say forget it and put part of the instructions for making a new MVC project downloading this extension and using it to move the project folders to the main folder.

**Comment by taori** [comment]
Yeah. You had the same thought process as me then. I don't know how urgent your thought of getting this to work with my extension but perhaps in your case it is worth to also look at dotnets new templating system. 

See [this](https://www.youtube.com/watch?v=GDNcxU0_OuE)

I imagine you need it quick and the issue is i won't get around to fixing it that soon, because i am too busy and runtime binding testing is usually a time consuming topic.

When i looked at the new templating system i thought "too bad i wasted all the time on writing this" - so perhaps its worth taking a look at. You can also add a meta reference to get intellisense when writing dotnet templates, which can be used for pretty much anything you like. That has the additional benefit that it has far less runtime dependencies and will work with ongoing versions of VS without me having to update the extension.

**Comment by angelicochris** [comment]
> 
> 
> Yeah. You had the same thought process as me then. I don't know how urgent your thought of getting this to work with my extension but perhaps in your case it is worth to also look at dotnets new templating system.
> 
> See [this](https://www.youtube.com/watch?v=GDNcxU0_OuE)
> 
> I imagine you need it quick and the issue is i won't get around to fixing it that soon, because i am too busy and runtime binding testing is usually a time consuming topic.
> 
> When i looked at the new templating system i thought "too bad i wasted all the time on writing this" - so perhaps its worth taking a look at. You can also add a meta reference to get intellisense when writing dotnet templates, which can be used for pretty much anything you like. That has the additional benefit that it has far less runtime dependencies and will work with ongoing versions of VS without me having to update the extension.

That is only for .Net Core projects though right? Unfortunately we are not doing core yet either.

**Comment by taori** [comment]
Nope. Not just .net core projects either. It is not language agnostic. Technically you could use it to make folder structures completely independent of what projects you want to scaffold.

It is a very generic template engine capable of generating whole hierarchies of folders.

the only "dependency" would be having the dotnetcore sdk installed so you can then do stuff like "dotnet new mycustomtemplate". The benefit of using this is you can also add parameterized scaffolding in your template, like excluding folders and stuff. 

Take a look at the link really. I think in the future that approach will be the way to go either way and since at some point you will transition to VS2019 you will have the added benefit of wizard parameters beyond just telling the wizard which project name should be used. Personally i would not bother with old templating anymore. The new version is just too good + flexible really. Not as convenient as doing with a WPF application on top of it, yes - but much more versatile.

Since the netcore sdk would be only required for scaffolding+building the template i think its the simplest and most flexible way, unless you cannot install programs on a developer machine... but then again. that would be really really odd for a developer machine
